Aggiunge capitolo Cosa significa essere dev (#111)
Il-Libro-Open-Source/book: docs/cosa-significa-essere-un-dev/README.md

> # Cosa significa essere un dev
> 

Reviewer: ```suggestion
```

Rimosso il titolo in quanto non necessario.

> _Cosa significa "essere un dev"?_

Reviewer: ```suggestion
_Cosa significa "essere dev"?_
```

> La domanda, di per se, è piuttosto semplice.

Reviewer: ```suggestion
La domanda, di per sé, è piuttosto semplice.
```

> Non si può dire però altrettanto per la risposta...o, per meglio dire, le risposte.

Reviewer 2: Affermativo

> Non si può dire però altrettanto per la risposta...o, per meglio dire, le risposte.

Reviewer: ```suggestion
Non si può dire però altrettanto per la risposta... o, per meglio dire, le risposte.
```

> Ebbene si, essere dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere "_Essere dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"... ma sarebbe riduttivo.

Reviewer 3: ```suggestion
Ebbene sì, essere dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere "_Essere dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"... ma sarebbe riduttivo.
```

> Ebbene si, essere un dev và oltre una singola e banale definizione: a tale domanda si potrebbe rispondere con un "_Essere un dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"...ma sarebbe riduttivo.

Reviewer 2: "va", essendo terza persona singolare!

> Ebbene si, essere un dev và oltre una singola e banale definizione: a tale domanda si potrebbe rispondere con un "_Essere un dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"...ma sarebbe riduttivo.

Reviewer: ```suggestion
Ebbene si, essere un dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere con un "_Essere un dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"...ma sarebbe riduttivo.
```

Reviewer 4: Manca uno spazio dopo i tre punti

> Ebbene si, essere un dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere con un "_Essere un dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"... ma sarebbe riduttivo.

Reviewer: @serenasensini "essere un dev" o "essere dev"?

Author: Dalla sua risposta dipendono anche altri parti (incluso il titolo del capitolo) 😄

> Ebbene si, essere un dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere con un "_Essere un dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"... ma sarebbe riduttivo.

Reviewer: ```suggestion
Ebbene si, essere dev va oltre una singola e banale definizione: a tale domanda si potrebbe rispondere "_Essere dev significa ricoprire una figura professionale dedita alla stesura del codice e con il fine ultimo di creare prodotti digitali_"... ma sarebbe riduttivo.
```

> Oppure si potrebbe dire che "_Essere un dev significa essere persone altamente competenti nel campo informatico; figure prettamente tecniche e verticalizzate in uno o più linguaggi di programmazione_"...ma anche qui, **sarebbe come indossare un maglione troppo stretto**: dopo un po' pizzica.

Reviewer: ```suggestion
Oppure si potrebbe dire che "_Essere dev significa essere persone altamente competenti nel campo informatico; figure prettamente tecniche e verticalizzate in uno o più linguaggi di programmazione_"...ma anche qui, **sarebbe come indossare un maglione troppo stretto**: dopo un po' pizzica.
```

> Potremmo continuare dicendo che "_Essere un dev significa essere un'artigiano del mondo digital: un professionista in grado di trovare la soluzione ad uno specifico problema ed essere in grado di applicarla in un contesto digitale_".

Reviewer 2: "va" sempre senza accento

> Potremmo continuare dicendo che "_Essere un dev significa essere un'artigiano del mondo digital: un professionista in grado di trovare la soluzione ad uno specifico problema ed essere in grado di applicarla in un contesto digitale_".

Reviewer: ```suggestion
Potremmo continuare dicendo che "_Essere un dev significa essere un artigiano del mondo digital: un professionista in grado di trovare la soluzione ad uno specifico problema ed essere in grado di applicarla in un contesto digitale_".
```

> Potremmo continuare dicendo che "_Essere un dev significa essere un artigiano del mondo digital: un professionista in grado di trovare la soluzione ad uno specifico problema ed essere in grado di applicarla in un contesto digitale_".

Reviewer: ```suggestion
Potremmo continuare dicendo che "_Essere dev significa essere artigiani del mondo digital: un professionista in grado di trovare la soluzione ad uno specifico problema ed essere in grado di applicarla in un contesto digitale_".
```

> O ancora, volendola vedere sotto un punto di vista più etico: "_Essere un dev significa abbracciare il movimento Open Source, apportando il proprio contributo senza pretendere nulla in cambio e contribuendo così a far crescere la community stessa, oltre che il proprio know-how_".

Reviewer: ```suggestion
O ancora, volendola vedere sotto un punto di vista più etico: "_Essere dev significa abbracciare il movimento Open Source, apportando il proprio contributo senza pretendere nulla in cambio e contribuendo così a far crescere la community stessa, oltre che il proprio know-how_".
```

> La verità, caro lettore, è che essere un dev significa essere tante cose contemporaneamente.

Reviewer: ```suggestion
La verità, caro lettore, è che essere dev significa essere tante cose contemporaneamente.
```

Tante _definizioni_, se così vogliamo chiamarle, che però affondano le loro radici su dei principi solidi e genuini: [**passione**](#la-passione-il-combustibile-per-eccellenza), [**curiosità**](#la-curiosità-il-potere-della-scoperta), [**empatia**](#lempatia-la-capacità-di-impersonare-il-prossimo) e [**contaminazione**](#la-contaminazione-fare-squadra-nel-modo-giusto); principi su cui, peraltro, nasce questo libro.

Caratteristiche che nulla hanno a che fare con la scrittura del codice, né presentano alcun tipo di legame diretto con specifiche competenze tecniche; nonostante ciò, risultano essere caratteristiche **fondamentali** per intraprendere questa carriera al meglio, lasciarsi coinvolgere da essa e riuscire a portare valore in questo settore.

## La passione: il combustibile per eccellenza

> Decidere di "diventare un dev" significa scegliere di "sacrificare" buona parte del proprio tempo al fine di raggiungere un'obiettivo; non sarà un percorso facile, tantomeno breve; ci saranno momenti di eccitazione, ma anche di sconforto; ci saranno giorni in cui tutto sembrerà andare per il verso giusto, ed altri in cui ogni nostra singola attività verrà minata dalle difficoltà che ci si pareranno davanti; ci saranno concetti che domeremo con estrema facilità, ed altri in cui ne usciremo sconfitti e con la coda tra le gambe.

Reviewer: Al posto di "buona parte" proporrei un più modesto "parte". Sono d'accordo sul punto ma non necessariamente il proprio tempo dovrà essere "comandato" dallo sviluppo, quindi un "parte" più "moderato" mi suona meglio, che ne pensi?

> Decidere di "diventare un dev" significa scegliere di "sacrificare" buona parte del proprio tempo al fine di raggiungere un'obiettivo; non sarà un percorso facile, tantomeno breve; ci saranno momenti di eccitazione, ma anche di sconforto; ci saranno giorni in cui tutto sembrerà andare per il verso giusto, ed altri in cui ogni nostra singola attività verrà minata dalle difficoltà che ci si pareranno davanti; ci saranno concetti che domeremo con estrema facilità, ed altri in cui ne usciremo sconfitti e con la coda tra le gambe.

Reviewer 2: "essere un dev" sostituibile con "lavorare come dev" o "diventare dev"

> Decidere di "diventare un dev" significa scegliere di "sacrificare" buona parte del proprio tempo al fine di raggiungere un'obiettivo; non sarà un percorso facile, tantomeno breve; ci saranno momenti di eccitazione, ma anche di sconforto; ci saranno giorni in cui tutto sembrerà andare per il verso giusto, ed altri in cui ogni nostra singola attività verrà minata dalle difficoltà che ci si pareranno davanti; ci saranno concetti che domeremo con estrema facilità, ed altri in cui ne usciremo sconfitti e con la coda tra le gambe.

Reviewer: ```suggestion
Decidere di "diventare dev" significa scegliere di "sacrificare" parte del proprio tempo al fine di raggiungere un obiettivo; non sarà un percorso facile, tantomeno breve; ci saranno momenti di eccitazione, ma anche di sconforto; ci saranno giorni in cui tutto sembrerà andare per il verso giusto, ed altri in cui ogni nostra singola attività verrà minata dalle difficoltà che ci si pareranno davanti; ci saranno concetti che domeremo con estrema facilità, ed altri in cui ne usciremo sconfitti e con la coda tra le gambe.
```

> Fà tutto parte del gioco: _abbraccialo, comprendilo e fallo tuo_; magari non oggi e forse nemmeno domani, ma arriverà il momento in cui riuscirai dove prima hai fallito.

Reviewer: ```suggestion
Fa tutto parte del gioco: _abbraccialo, comprendilo e fallo tuo_; magari non oggi e forse nemmeno domani, ma arriverà il momento in cui riuscirai dove prima hai fallito.
```

> Solitamente, quando ci appassioniamo a qualcosa tendiamo a dedicare molto del nostro tempo a quella specifica attività; la coltiviamo negli anni prendendocene cura, approfondendo ogni aspetto e cercando di scoprire ogni singola caratteristica che possiede...in buona sostanza: _la rendiamo nostra_.

Reviewer: Trasformerei i tre puntini in un ";" con uno spazio dopo

Reviewer 4: Io sono piu propenso per l'uso del punto piuttosto che del punto e virgola. Cosi la "responsabilita" della frase rimane una sola. E non tre. In oltre il punto ferma la frase li e da un gran peso a quel che si legge.

Reviewer: Anche, sì!

Reviewer 2: Dal mio punto di vista, è una questione di scrittura e di stile. Un po' come ogni autore/autrice ha la sua penna preferita, qui si tratta di scegliere anche il modo in cui lo si vuole leggere e soprattutto come si vuole che venga letto. In realtà, sono corrette tutte e tre le forme dal punto di vista sintattico, anche se, riconoscendo il tono "romantico" e informale del capitolo, i tre puntini (senza abusarne), ci stanno!

Forti di questa consapevolezza possiamo quindi affermare che ciò che farà la differenza tra il nostro successo e la nostra sconfitta sarà la passione che metteremo in ciò che facciamo; se non avremo passione, difficilmente riusciremo ad accettare una sconfitta e a rialzarci più forti di prima; senza passione non riusciremo a trovare la forza di studiare, magari dopo una giornata di lavoro.

Se però avremo passione, avremo dalla nostra una risorsa sorprendente: il dono della _Perseveranza_.

> Non importerà quindi quante volte proveremo, senza successo, a completare un'esercizio o ad imparare una tecnologia: avremo sempre la forza necessaria per incassare il colpo, prendere fiato, e provare a superare noi stessi.

Reviewer: ```suggestion
Non importerà quindi quante volte proveremo, senza successo, a completare un esercizio o ad imparare una tecnologia: avremo sempre la forza necessaria per incassare il colpo, prendere fiato, e provare a superare noi stessi.
```

> Il mio consiglio per te caro lettore, è quello di tenere a mente, per l'intera durata di questo manoscritto, il ruolo che ricopre la passione riguardo al percorso che hai scelto di intraprendere: se avrai passione, avrai la forza di perseverare.

Reviewer: Toglierei "per l'intera durata di questo manoscritto". La frase mi piace molto, ma non sappiamo ancora in che punto del libro sarà questo contenuto. Inoltre, gli argomenti sono vari e disparati quindi non è detto vengano letti nell'ordine che ci aspettiamo. Se questo capitolo fosse, che ne so, il penultimo, la frase assumerebbe poco senso.

Reviewer 2: "caro lettore" sostituibile con "che leggi"

> Il mio consiglio per te caro lettore, è quello di tenere a mente, per l'intera durata di questo manoscritto, il ruolo che ricopre la passione riguardo al percorso che hai scelto di intraprendere: se avrai passione, avrai la forza di perseverare.

Reviewer: ```suggestion
Il mio consiglio per te che leggi, è quello di tenere a mente, per l'intera durata di questo manoscritto, il ruolo che ricopre la passione riguardo al percorso che hai scelto di intraprendere: se avrai passione, avrai la forza di perseverare.
```

_Persevera con passione: a quel punto l'unico limite che avrai, sarai tu._

## La curiosità: il potere della scoperta

Qualità caratteristica dell'essere umano, la curiosità è l'impulso a scoprire, ricercare e comprendere ciò che ancora non conosciamo. Se cerchiamo la definizione di "curiosità" su Wikipedia infatti, scopriamo che:

> _La curiosità è un aspetto emozionale che descrive un numero di meccanismi e comportamenti che hanno come fine il placare l'impulso degli esseri viventi a trarre informazioni ed interagire con l'ambiente._

> Il percorso che hai deciso di intraprendere pullula di informazioni che non aspettano altro che essere scoperte, studiate e messe in pratica; ci teniamo però a preparti: alcune nozioni potranno risultarti piuttosto semplici, altre dannatamente complesse.

Reviewer: "preparti"

> Il percorso che hai deciso di intraprendere pullula di informazioni che non aspettano altro che essere scoperte, studiate e messe in pratica; ci teniamo però a preparti: alcune nozioni potranno risultarti piuttosto semplici, altre dannatamente complesse.

Reviewer: ```suggestion
Il percorso che hai deciso di intraprendere pullula di informazioni che non aspettano altro che essere scoperte, studiate e messe in pratica; ci teniamo però a prepararti: alcune nozioni potranno risultarti piuttosto semplici, altre dannatamente complesse.
```

> Nonostante ciò, non scoraggiarti; la nostra intenzione non è quella di spaventarti, ma di consigliarti e guidarti nel migliore dei modi; proprio per questo, ti inviatiamo a riflettere su un'aspetto tanto scontato quanto potente: il potere dell'informazione libera.

Reviewer: ```suggestion
Nonostante ciò, non scoraggiarti; la nostra intenzione non è quella di spaventarti, ma di consigliarti e guidarti nel migliore dei modi; proprio per questo, ti inviatiamo a riflettere su un aspetto tanto scontato quanto potente: il potere dell'informazione libera.
```

> Proprio come il libro che stai leggendo in questo momento, in rete esistono miliardi di informazioni a cui puoi avere accesso: aspettano solo di essere ricercate! Queste pagine che stai leggendo sono presenti sul web, e molto probabilmente ne sei venuto a conoscenza perché ti sei imbattuto in esse navigando per la rete, o magari perché chi ti ha consigliato questo libro lo ha cercato a sua volta. Poco importa, quello che è davvero importante è che tu prenda consapevolezza del fatto che il web pullula di informazioni a cui puoi accedere, e che l'unica cosa che ti separa da esse, indovina un po', _sei tu_!

Reviewer: "Proprio come il libro che stai leggendo in questo momento" lo sostituirei con "Proprio come questo libro", perché dopo citi ancora "queste pagine che stai leggendo" e può risultare ripetitivo.

> Le informazioni presenti in rete però, per quanto siano molteplici e liberamente fruibili, da sole non hanno alcun effetto sulla tua formazione; analogamente a quanto avviene con un esplosivo, la polvere da sparo, per quanto potente possa essere, da sola non basta; serve un'innesco o, per meglio dire, serve **una scintilla che sia in grado di innescare la miccia**.

Reviewer 2: "ne sei venuto a conoscenza" con "ne sei a conoscenza"
"ti sei imbattuto in esse" sostituibile con "è stato semplice imbattersi in esse"

> A questo punto, caro lettore, avrai capito dove vogliamo andare a parare: ciò che non deve mai mancare in te è la curiosità!

Reviewer 2: "caro lettore" sostituibile con "a te che leggi"

Forse ti starai chiedendo cosa c'entri una qualità come l'empatia in un percorso di apprendimento e crescita professionale in un settore come quello della programmazione.

Se ad un primo impatto potrebbe sembrare un dubbio legittimo, la verità è che l'empatia in questo caso gioca un ruolo fondamentale.

Durante il tuo percorso, presto o tardi ti capiterà di imbatterti in un problema a cui non saprai far fronte; a quel punto, molto probabilmente il tuo primo impulso sarà quello di cercare in rete una soluzione e, pensa un po', la troverai!

Penserai "_Che fortuna che ho avuto!_", ed è qui che sbaglieresti...**sbaglieresti di grosso**.

Quella soluzione, che molto probabilmente ti aiuterà a risolvere il tuo problema, è lì per un motivo ben preciso: qualcuno, prima di te, ha avuto il tuo stesso problema; al contrario di te però, quel qualcuno non è stato "_così fortunato_" da trovare una soluzione online, ma è stato abbastanza empatico da mettersi nei panni di chi, come te, quel problema prima o poi lo avrebbe avuto; una volta trovata la soluzione quindi, si è adoperato per metterla a disposizione di tutti, **gratuitamente** e in maniera del tutto spontanea.

> Questo, caro lettore, è uno dei principi fondamentali su cui verte questo libro: **la condivisione libera del sapere (in questo caso collettivo) con chiunque ne abbia bisogno**.

Reviewer 2: sostituibile con "a te che leggi"

> Questo, caro lettore, è uno dei principi fondamentali su cui verte questo libro: **la condivisione libera del sapere (in questo caso collettivo) con chiunque ne abbia bisogno**.

Reviewer: ```suggestion
Questo, a te che leggi, è uno dei principi fondamentali su cui verte questo libro: **la condivisione libera del sapere (in questo caso collettivo) con chiunque ne abbia bisogno**.
```

> Ed è proprio per questo che ti parlo di empatia: durante il tuo percorso, che sia grazie a questo libro o ad altro, apprenderai un'enorme quantità di informazioni; mentre lo farai, rifletti sul perché ti è possibile farlo, e domandati se le nozioni da te apprese potranno in futuro essere utili a chi muoverà i primi passi in questo settore, proprio come stai facendo tu ora grazie al manoscritto che stai leggendo.

Reviewer 2: Ci può stare la virgola, nel caso si voglia dare enfasi alla frase :)

Forte di questa consapevolezza, prendi coraggio e sii empatico: pubblica, ovunque tu voglia e in qualsiasi formato tu preferisca, ciò che hai imparato: per te sarà un notevole esercizio di ripasso e pratica sul campo; chiunque fruirà dei tuoi contenuti te ne sarà grato e sarà propenso a diventare un membro della tua community, ma sopra ogni cosa sarà ispirato a condividere a sua volta le nozioni che imparerà lungo il suo percorso...proprio come hai fatto tu prima di lui.

> Ciò che vogliamo trasmetterti è la propensione ad adottare uno spirito di condivisione: oltre ad essere il cuore pulsante del movimento Open Source, e quindi del libro che stai leggendo, è anche un principio nobile e al contempo stesso utile; qualora non ne comprenderai subito il senso ed il potenziale, o magari ti sembrerà una perdita di tempo, tieni a mente quanto segue: arriverà il giorno in cui ringrazierai chi, prima di te, ha seguito il nostro consiglio.

Reviewer 2: metterei in grassetto, per evidenziare l'importanza del concetto

Author: @serenasensini Intendi tutto il paragrafo?

> Fidati di noi: fidati di chi, prima di te, è stato aiutato e che oggi è qui a condividere il suo sapere con te.

Reviewer: Questi ultimi 4 paragrafi, presi singolarmente e anche nel loro insieme, mi piacciono molto, ma non vorrei risultassero troppo ripetitivi. È più una domanda che pongo a chi leggerà queste parole e questa review: Si sta dando enfasi al concetto in maniera corretta o "un pochino troppo"? Ripeto, non è la scelta delle parole, che approvo al 100%; ma alla ripetizione del concetto.

Reviewer: Rileggendolo credo sia sufficiente rimuovere il "fidati di noi.... con te" per non esasperare il concetto.

## La contaminazione: fare squadra nel modo giusto

> "Essere un dev", tra le tante altre cose, significa _essere parte di una comunità_.

Reviewer 2: Sostituibile con "indossare i panni di dev", che ne pensi?

Una comunità composta da altri professionisti, persone che come te hanno scelto di lavorare in questo settore.

Con il tempo, magari collaborando ad un progetto Open Source o lavorando in qualche azienda, ti troverai a dover collaborare a stretto contatto (che sia esso virtuale o fisico) con altri sviluppatori.

Scenario ben diverso da ciò che magari sei stato abituato a fare fino a quel momento; d'altro canto, un conto è scrivere codice in solitudine per un progetto a cui stiamo lavorando, e un conto è scrivere codice collaborativo con un numero indefinito di altre persone, tutte con lo stesso obiettivo.

> In queste circostante, ciò che solitamente fa la differenza è lo spirito con cui si affrontano queste situazioni; partire prevenuti non serve a niente, e anzi il più delle volte potrebbe addirittura risultare controproducente; ha molto più senso invece partire dall'assunto che **ognuno di noi può insegnarci qualcosa**, e dunque, **possiamo sempre imparare qualcosa, anche da chi è meno esperto di noi**.

Reviewer: ```suggestion
In queste circostanze, ciò che solitamente fa la differenza è lo spirito con cui si affrontano queste situazioni; partire prevenuti non serve a niente, e anzi il più delle volte potrebbe addirittura risultare controproducente; ha molto più senso invece partire dall'assunto che **ognuno di noi può insegnarci qualcosa**, e dunque, **possiamo sempre imparare qualcosa, anche da chi è meno esperto di noi**.
```

> In queste circostante, ciò che solitamente fà la differenza è lo spirito con cui si affrontano queste situazioni; partire prevenuti non serve a niente, e anzi il più delle volte potrebbe addirittura risultare controproducente; ha molto più senso invece partire dall'assunto che **ognuno di noi può insegnarci qualcosa**, e dunque, **possiamo sempre imparare qualcosa, anche da chi è meno esperto di noi**.

Reviewer 2: Confermo "fa", terza persona singolare. "Fa'" apostrofato è per l'imperativo, "fà" è sempre sbagliato!

> Adottare un'atteggiamento del genere significa abbracciare l'idea di poter apprendere sempre qualcosa, indipendentemente dal nostro grado di conoscenza o da chi abbiamo davanti...significa **lasciarsi contaminare**.

Reviewer: ```suggestion
Adottare un atteggiamento del genere significa abbracciare l'idea di poter apprendere sempre qualcosa, indipendentemente dal nostro grado di conoscenza o da chi abbiamo davanti...significa **lasciarsi contaminare**.
```

> La contaminazione, in un gruppo di persone aventi lo stesso obiettivo, è un'atteggiamento prono al successo, è inclusivo e privo di qualsiasi pregiudizio di genere: è un modo sano di fare gruppo, contribuendo alla sua crescita e apprendendo da esso.

Reviewer 5: L'ultimo periodo mi sembra poco chiaro, potrebbe confondersi con la contaminazione invece che con il gruppo. Che ne pensi di rielaborarlo?

Author: @Il-Libro-Open-Source/ambassadors Anche qui, voi cosa ne pensate? Idee?

> In fondo, come abbiamo già detto qualche riga fà, Open Source significa anche questo: contribuire condividendo il proprio sapere, ed apprendere da chi a sua volta condivide il suo sapere con noi.

Reviewer 5: Nit: fa al posto di fà

> In fondo, come abbiamo già detto qualche riga fà, Open Source significa anche questo: contribuire condividendo il proprio sapere, ed apprendere da chi a sua volta condivide il suo sapere con noi.

Reviewer: ```suggestion
In fondo, come abbiamo già detto qualche riga fa, Open Source significa anche questo: contribuire condividendo il proprio sapere, ed apprendere da chi a sua volta condivide il suo sapere con noi.
```
